app/controllers/messages_controller.rb

```diff
@@ -3,7 +3,9 @@ class MessagesController < ApplicationController
   def create
     @case = Case::Base.find params[:case_id]
     authorize(@case, :can_add_message_to_case?)
-    team = current_user.teams_for_case(@case).first
+    teams = current_user.teams_for_case(@case)
+    weightings = { 'manager' => 100, 'approver' => 200, 'responder' => 300 }
+    team = teams.sort{ |a, b| weightings[a.role] <=> weightings[b.role] }.first
```

Comment:
What I don't understand here is that in the [Case::BasePolicy](https://github.com/ministryofjustice/correspondence_tool_staff/blob/ae66f53af2483eea8757a449633b0d86851d7d7e/app/policies/case/base_policy.rb#L270), it doesn't matter which of these roles the user has, so long as it's not closed. Is this policy now different in the configurable state machine config? If it is, isn't the simplest and most reasonable temporary fix to change the config to match?

Reply:
The config state machine only allows managers to add messages to a case in unassigned state, and that's right, no?  Sure, I could add in config to allow anybody to add a message, but that is covering up the issue, rather than sorting it out.

Reply:
I'm not sure that we want to prevent people from commenting on the case while it's unassigned ... it's possible that that behavior might simply emerge from how people currently work with the system, but is there a reason we want to explicitly prevent it? If anything, I would expect that we want to allow anyone to comment on a case, so long as they have permissions to see it at all.

In any case, I think that this brings up the point that we should probably try to replace like for like. I know I've been guilty of trying to "improve" behaviour when making other changes to the system, but we really need to be careful of that and, probably, avoid that where possible.